kyo-data/shared/src/main/scala/kyo/Duration.scala

```diff
@@ -174,13 +174,24 @@ object Duration:
                 case Duration.Zero => JavaDuration.ZERO
                 case n             => JavaDuration.of(n.toNanos, NANOS)
 
-        /** Converts the Duration to a human-readable string.
+        /** Converts the Duration to a human-readable string at the most coarse possible resolution without losing information.
           *
           * @return
           *   A string representation of the Duration
           */
-        inline def show: String = s"Duration($self ns)"
-
+        def show: String =
+            if self == Zero then "Duration.Zero"
+            else if self == Infinity then "Duration.Infinity"
```

Comment:
Is there also `-Infinity`?

Reply:
no, the current impl doesn't allow negative durations